### PR TITLE
chore(scanner): cleanup client-side OS conversion

### PR DESCRIPTION
**Pull request by RTann**
## Description

**NOTE: Original purpose of PR will be implemented in https://github.com/stackrox/stackrox/pull/9345. This PR now just does simple cleanup**

I noticed `VersionId` is not populated for Alpine. See https://github.com/quay/claircore/blob/v1.5.20/alpine/distributionscanner.go#L110

I also checked the Scanner v4 DB and saw the following:

```
# SELECT did, version_id, version FROM dist;
  did   | version_id |     version     
--------+------------+-----------------
 rhel   | 9          | 9
 rhel   | 8          | 8
 debian | 12         | 12 (bookworm)
 debian | 11         | 11 (bullseye)
 ubuntu | 22.04      | 22.04 (Jammy)
 alpine |            | 3.18
 alpine |            | 3.16
 alpine |            | 3.17
 alpine |            | 3.11
 alpine |            | 3.8
 alpine |            | 3.14
 debian | 9          | 9 (stretch)
 alpine |            | 3.10
 alpine |            | 3.9
 alpine |            | 3.12
 alpine |            | 3.15
 ubuntu | 20.04      | 20.04 (Focal)
 alpine |            | 3.19
 alpine |            | 3.13
 alpine |            | 3.2
 ubuntu | 18.04      | 18.04 (Bionic)
 debian | 10         | 10 (buster)
 ubuntu | 14.04      | 14.04 (Trusty)
 ubuntu | 16.04      | 16.04 (Xenial)
 rhel   | 7          | 7
 debian | 8          | 8 (jessie)
 ubuntu | 21.10      | 21.10 (Impish)
 debian | 7          | 7 (wheezy)
 ubuntu | 22.10      | 22.10 (Kinetic)
 amzn   | 2018.03    | 2018.03
 alpine |            | 3.7
 ubuntu | 23.10      | 23.10 (Mantic)
 ubuntu | 23.04      | 23.04 (Lunar)
(33 rows)
```

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

### Here I tell how I validated my change

CI

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Review by jvdm — COMMENTED** [review]
I think this should be done inside Scanner, wrapping ClairCore, which (should) eventually fix this: https://github.com/stackrox/stackrox/pull/9345

File: pkg/scanners/scannerv4/convert.go [hunk 1]
```diff
@@ -186,9 +186,17 @@ func normalizedSeverity(severity v4.VulnerabilityReport_Vulnerability_Severity)
 // If there are zero known distributions for the image or if there are multiple distributions,
 // return "unknown", as StackRox only supports a single base-OS at this time.
 func os(report *v4.VulnerabilityReport) string {
-	if len(report.GetContents().GetDistributions()) == 1 {
-		for _, dist := range report.GetContents().GetDistributions() {
-			return dist.Did + ":" + dist.VersionId
+	dists := report.GetContents().GetDistributions()
+	if len(dists) == 1 {
+		for _, dist := range dists {
```
[the remark this answers is not in the document]
Comment (RTann) [reply pkg/scanners/scannerv4/convert.go:191]: good point I think this was originally written assuming `dists` is a map

**Comment by RTann** [comment]
Replacing most of this work with https://github.com/stackrox/stackrox/pull/9345 but keeping this PR for cleanup

**Review by jvdm — APPROVED** [review]
LGTM.

**Pull request merged**